documentation/src/main/asciidoc/reference/search-dsl-predicate.asciidoc

```diff
@@ -67,6 +67,19 @@ include::{sourcedir}/org/hibernate/search/documentation/search/predicate/Predica
 * The score of a `matchAll` predicate is constant and equal to 1 by default,
 but can be <<search-dsl-predicate-common-boost,boosted with `.boost(...)`>>.
 
+[[search-dsl-predicate-match-none]]
+== `matchNone`: match no documents
+
+The `matchNone` predicate is the inverse of `matchAll` and matches no documents.
+
+.Matching no documents
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=matchNone]
+----
+====
+
```

Comment:
Sorry, I didn't include the original use case in the ticket; here it is: https://discourse.hibernate.org/t/fail-fast-predicate/6062?u=yrodiere

The best solution to avoid running a search query is simply to not call `.fetch()` at all, but in some cases people will have methods dedicated to the creation of inner predicates, and will not be able to prevent the query to run; in that case, they can just create a `matchNone` predicate and the end result will be the same (though it will unfortunately involve some pointless I/O).

So, as you can see, it's a very specific use case, for advanced users. I'm not sure we need further explanation than what you did here, but if you think you can add something clear and concise, feel free to give it a shot :)

Reply:
Hmm, that's an interesting case...  Knew of `match_all` usage within dynamic filters but not the `match_none`. nice :smiley: 
As for the text, I gave it a try ... aaaand the text becomes too cumbersome :smile: but as you've said - it's for advanced users. So those who need it would know its meaning, and others are probably safer without using it :smiley: